### PR TITLE
[MINOR][DOCS] Typo fixes in Dataset scaladoc

### PR DESCRIPTION
**Pull request by xinhhuynh**
## What changes were proposed in this pull request?

Minor typo fixes in Dataset scaladoc
- Corrected context type as SparkSession, not SQLContext.
  @liancheng @rxin @andrewor14 
## How was this patch tested?

Compiled locally

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -1665,7 +1665,7 @@ class Dataset[T] private[sql](
   }
 
   /**
-   * Returns a new [[Dataset]] by adding a column with metadata.
+   * Returns a new [[DataFrame]] by adding a column with metadata.
```
Comment (andrewor14) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: these were technically correct; a `DataFrame` is a `Dataset[Row]`. I would revert these changes

Comment (xinhhuynh) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: OK, I'll revert those.

**Comment by liancheng** [comment]
test this please

**Comment by liancheng** [comment]
LGTM pending Jenkins. Thanks!

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 2]
```diff
@@ -133,7 +133,7 @@ private[sql] object Dataset {
  *
  * and in Java:
  * {{{
- *   // To create Dataset<Row> using SQLContext
+ *   // To create Dataset<Row> using SparkSession
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: I think the problem here is space at the end of the line

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
This looks OK, but in the future @xinhhuynh it's better to include more changes even in a minor patch so there's less review overhead.

**Comment by andrewor14** [comment]
Merging into master 2.0
